### DISCUSSION TITLE
HTML form submit vs. action

### DISCUSSION THREAD

**Discussion by taalas** [livewire/livewire#4972] [1. Help]
Hi!

Sorry, this question might be trivial (and also a more general than a livewire question), but I am currently following the screencast series and was wondering:

When creating a frontend for modifying an existing object, you would bind the object's properties to livewire properties and then generate an html form with the corresponding input elements etc. This form can then be submitted, the contents validated and finally written back to persistence.

Is this equivalent to just use the input elements without grouping them in a form, then haven the user activate saving through a button that calls a method through an action. This action then reads the livewire properties, validates and writes them back in to the object.

What are the pros and cons?

**Comment by DGuttmanSN** [comment]
Submit is used when you need to send the form to the back application (php, java ...) and the action is only for execute methods or events in scripts

**Comment by joshhanley** [comment]
@taalas it's generally a good practice to wrap any inputs in a form and add `wire:submit.prevent` to the form instead of saving via an action button, as it has support for accessibility by allowing a user to submit the form by pressing the enter key when focused on an input.

Have a look at this video for more details https://laravel-livewire.com/screencasts/form-submits

Hope this helps!
